packages/core/src/Resource/Change/Relation/Relation.js

```diff
@@ -64,3 +64,26 @@ export function getChangedResourceRelations(resource) {
     ),
   ));
 }
+
+/**
+ * @param {HyralResource} resource
+ *
+ * @returns {HyralResource[]}
+ */
+export function getDeletedOneToOneRelatedResources(resource) {
+  if (!resource.relationships) {
+    return [];
+  }
+
+  if (resourceIsNew(resource) || previousState(resource.stateStack) === null) {
```

Comment:
what's the reason why you got these two checks together., but a seperate one on line 74?

maybe for consistency just seperate checks to keep it more readable?

Reply:
hmm, good point, was a copy from another function. Let me see if I can refactor it

Reply:
refactored code to use existing function to check if a resource has changed. Also refactored the original code and updated if statements to a single check.